sources/engine/Stride.Assets/Entities/ComponentChecks/RequiredMembersCheck.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Linq;
+using System.Reflection;
+using Stride.Core.Annotations;
+using Stride.Core.Assets;
+using Stride.Core.Assets.Compiler;
+using Stride.Core.Reflection;
+using Stride.Engine;
+
+namespace Stride.Assets.Entities.ComponentChecks
+{
+    /// <summary>
+    /// Checks if all members of a component with a <see cref="MemberRequiredAttribute"/> are assigned a value.
+    /// </summary>
+    public class RequiredMembersCheck : IEntityComponentCheck
+    {
+        /// <inheritdoc/>
+        public bool AppliesTo(Type componentType)
+        {
+            return true; // applies to any component
+        }
+
+        /// <inheritdoc/>
+        public void Check(EntityComponent component, Entity entity, AssetItem assetItem, string targetUrlInStorage, AssetCompilerResult result)
+        {
+            var typeDescriptor = TypeDescriptorFactory.Default.Find(component.GetType());
+            var componentName = typeDescriptor.Type.Name; // Should we check attributes for another name?
+
+            var members = typeDescriptor.Members;
+
+            foreach (var member in members)
+            {
+                if (member.Type.IsValueType)
+                    continue; // value types cannot be null, and must always have a proper default value
+
+                MemberRequiredAttribute memberRequired;
+                if ((memberRequired = member.GetCustomAttributes<MemberRequiredAttribute>(false).FirstOrDefault()) != null)
```

Comment:
Here I used `inherit: false` - if I understand it correctly, this should be `true` when inspecting overloaded methods/properties and looking at attributes across the inheritance tree, should I change it?

Reply:
Yes it should probably be true (in case it's declared on the `virtual` member but not on the `overridden` one)